diff -urN idlelib12.orig/configHandler.py idlelib12/configHandler.py
--- idlelib12.orig/configHandler.py	2006-10-14 18:17:04.000000000 +0700
+++ idlelib12/configHandler.py	2006-10-14 18:24:50.000000000 +0700
@@ -407,7 +407,7 @@
         names=extnNameList
         kbNameIndicies=[]
         for name in names:
-            if name.endswith(('_bindings', '_cfgBindings')):
+            if name.endswith('_bindings') or name.endswith('_cfgBindings'):
                 kbNameIndicies.append(names.index(name))
         kbNameIndicies.sort()
         kbNameIndicies.reverse()
diff -urN idlelib12.orig/PyShell.py idlelib12/PyShell.py
--- idlelib12.orig/PyShell.py	2006-10-14 18:17:04.000000000 +0700
+++ idlelib12/PyShell.py	2006-10-14 18:36:04.000000000 +0700
@@ -358,10 +358,9 @@
         # run from the IDLE source directory.
         del_exitf = idleConf.GetOption('main', 'General', 'delete-exitfunc',
                                        default=False, type='bool')
-        if __name__ == 'idlelib.PyShell':
-            command = "__import__('idlelib.run').run.main(%r)" % (del_exitf,)
-        else:
-            command = "__import__('run').main(%r)" % (del_exitf,)
+        package_path = __name__.split('.')[:-1]
+        run_path = '.'.join(package_path+['run'])
+        command = "__import__('%s').run.main(%r)" % (run_path, del_exitf, )
         if sys.platform[:3] == 'win' and ' ' in sys.executable:
             # handle embedded space in path by quoting the argument
             decorated_exec = '"%s"' % sys.executable
